test/sql/function/timestamp/test_strptime.test

```diff
@@ -72,6 +72,45 @@ SELECT strptime('jun', '%b')
 ----
 1900-06-01 00:00:00
 
+# First Monday is jan 1
+query I
+SELECT strptime('1', '%W')
+----
+1900-01-01 00:00:00
+
+# First Sunday is jan 7
+query I
+SELECT strptime('1', '%U')
+----
+1900-01-07 00:00:00
+
+# Ignored
+query I
+SELECT strptime('6', '%w')
+----
+1900-01-01 00:00:00
+
+query I
+SELECT strptime('30', '%U')
+----
+1900-07-29 00:00:00
+
+query I
+SELECT strptime('30', '%W')
+----
+1900-07-23 00:00:00
+
+# 4 days after Monday
+query I
+SELECT strptime('1-4', '%W-%w')
+----
+1900-01-05 00:00:00
+
+query I
+SELECT strptime('30', '%j')
```

Comment:
Could you add some more tests, e.g.:

* Could you test week/weekday together with a year?
* What happens if both month/day and week/weekday are specified?
* What happens if both month/day and week/weekday are specified, but there is a mismatch?
* Could you test a round-trip with strftime %W/%w for several dates (i.e. that the strftime and strptime are consistent with one another)?

Reply:
I'll put in the tests. The computations only depend on the year, and 1900 is a fun one (it isn't a leap year and it stars on a Monday), so I wasn't too worried. I manually checked some dates in this year, but I should have just added them as tests. the round trip is a good point.

The week/weekday vs month/day conflict is not checked, and basically the week/doy stuff will override. I can put in other checks, but the tricky part is that if there is no conflict, what then? (the mismatch case). Python seems very lax about it so I wasn't going to worry too much. 

There is also the performance angle - adding lots of checks will make it slower and the code as written seems to be trying to be fast. I was even feeling a bit guilty about adding the date arithmetic stuff for the week calculations, but there isn't really a way around that.

Reply:
How about a flag that says that mm/dd were specified, so just ignore the W/j specifiers (e.g., treat them like weekday names)?

Reply:
Hmm, there is an amusing round trip issue here because %w is the *Sunday* (%U) DOW but if you use it with the *Monday* (%W) week the result is off by one. I think the user would expect the round trip to work, so I will adjust the weekday to compensate when computing.

Reply:
Python round trips both, so that's what I think we should do.

Reply:
Ok the way it seems to work is that the weekday is that day of the week no matter what week start style is in use. So the weekday has to be adjusted to wrap. I've added test that check an entire year for discrepancies. Coming shortly.

Reply:
[Calendar for 1900](https://www.timeanddate.com/calendar/?year=1900)

Reply:
Thanks for the fixes! Looks great now. Never thought I would be interested in a calendar of the year 1900, but here we are ;)

Reply:
:1st_place_medal:

Reply:
1900 is a fun year because it a) is NOT a leap year (even though it is divisible by 4) and b) starts on a Monday. The original Mac toolbox used an epoch/zero value of 1904-01-01 just to avoid these issues. Then there was the old version of Excel that used 1900-01-01 and didn't understand the Gregorian calendar leap year rules...

Reply:
[The Excel issue was allegedly intentional by Microsoft for compatibility with Lotus 1-2-3 :)](https://docs.microsoft.com/en-us/office/troubleshoot/excel/wrongly-assumes-1900-is-leap-year)